### DISCUSSION TITLE
Sorting on Selected

### DISCUSSION THREAD

**Discussion by mjchang** [TanStack/table#2119] [General]
I'm trying to enable sorting on the select column and it **mostly** works.  When I manually toggle the column sort, the sorting works, but the sorting does not on initial render. This is because the row.isSelected is not populated until after useRowSelect runs.

Right now I'm thinking of using some useEffect to retrigger sort but this seems hacky and I have yet to come up with something that works. Anyone have any suggestions?

Codesandbox: https://codesandbox.io/s/competent-ellis-febbl?file=/src/App.js

**Comment by kachun333** [comment]
hi @mjchang, did you find a non-hacky solution?

Shall we open an issue to propose this as a new feature? 
Ability to sort by selected rows first/last

**Comment by anondawg21** [comment]
Any update for this?
